workflow/controller/artifact_gc.go

```diff
@@ -151,16 +147,16 @@ func (woc *wfOperationCtx) processArtifactGCStrategy(ctx context.Context, strate
 		}
 		templateName := node.TemplateName
 		if templateName == "" && node.GetTemplateRef() != nil {
-			templateName = node.GetTemplateRef().Name
+			templateName = node.GetTemplateRef().Template
 		}
```

Comment:
was accidentally referring to the WorkflowTemplate's Name here rather than the name of the template within the WorkflowTemplate